TD 2659 - Updates to wizard and file uploader

jegasriskantha opened the pull request:
contributes to  [TD-2569](https://sce.myjetbrains.com/youtrack/issue/TD-2659/Create-Step-2)

## Changes

- Adds new error prop to FileUploader to allow setting the FileUploader in an error state. 

Updates request by @Sowbhagya-ipg  here https://github.com/IPG-Automotive-UK/virto/pull/685#issuecomment-2064006311
- Updates FileUploader height to 56px from 70px
- Updates padding on wizard action.


## Dependencies

builds on this PR: https://github.com/IPG-Automotive-UK/react-ui/pull/844 as I needed a single pre-release to use in current development. 


## UI/UX
Error on FileUploader and new height:
![CleanShot 2024-04-18 at 21 39 03@2x](https://github.com/IPG-Automotive-UK/react-ui/assets/75164177/17e3e383-a4a8-4d59-920c-3fd7165a3ea1)

Padding on wizard actions: 
![CleanShot 2024-04-18 at 21 41 38@2x](https://github.com/IPG-Automotive-UK/react-ui/assets/75164177/3c7fd0d5-73e8-4e34-9ff4-36315f5fcd55)


## Testing notes

Automated tests:
- Added a test to check that when the error prop is set on the FileUploader the error styling is present. 

Manual testing:
- Check that the error prop sets the error styling on the file uploader. 

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [x] Branch has been run in docker.
- [x] PR assigned to me or an appropriate delegate.
- [x] Relevant labels added to the PR.
- [x] Appropriate tests have been added.
- [x] Lint and test workflows pass.

AronPetrauskas reviewed (commented):
Manually tested `FileUploader` in storybook, Confirmed padding of `WizzardActions`.

Left one comment regarding the functionality of the `error` prop on `FileUploader`. The logic might create a confusing behavior, just want to make sure this is the intended functionality.

jegasriskantha commented on src/FileUploader/FileUploader.tsx:
```diff
@@ -57,7 +58,7 @@ export default function FileUploader({
 
   // are we rendering an error state?
-  const isError = isDragReject || rejectionMessage;
+  const isError = isDragReject || rejectionMessage || error;
```
As discussed this is desired behaviour, but I have updated the comments to make this functionality a bit clearer.

AronPetrauskas reviewed (commented):
Good documentation, concise and clear 👍

Sowbhagya-ipg commented:
https://github.com/IPG-Automotive-UK/virto/pull/685#issuecomment-2064006311

I think there is a misunderstanding here @jegasriskantha The wizard I meant was actually the stepper on the top that has a bottom margin of 24px.  Attaching the image for ref.
<img width="780" alt="Screenshot 2024-04-22 at 10 34 00 am" src="https://github.com/IPG-Automotive-UK/react-ui/assets/114938041/1de2971d-da06-45e2-8db9-f53a08f74d9b">

jegasriskantha commented:
Reverted mistakingly changed margin, and updated the correct one: 
![CleanShot 2024-04-23 at 12 02 30@2x](https://github.com/IPG-Automotive-UK/react-ui/assets/75164177/aeac9e0a-297e-4ab1-8e5d-047eb181bf10)

AronPetrauskas commented:
Changes look good to me!
